### PR TITLE
feat: add nfs/smb share datasource and resource

### PR DESCRIPTION
**Pull request by lkubb**
**What kind of change does this PR introduce?**
* adds NFS share datasource and resource (+tests/go generate docs)
* adds SMB share datasource and resource (+tests/go generate docs)

**Other information**
Thanks for all the merges. 👍 Now everything is set and ready for this feature PR.
I tried to follow your coding style mostly, not only because this provider was my introduction to TF provider development in go. Please feel free to comment or suggest changes. :)

### CODE REVIEW

File: truenas/provider.go [hunk 1]
```diff
@@ -35,6 +35,8 @@ func Provider() *schema.Provider {
 		ResourcesMap: map[string]*schema.Resource{
 			"truenas_cronjob": resourceTrueNASCronjob(),
 			"truenas_dataset": resourceTrueNASDataset(),
+			"truenas_share_nfs": resourceTrueNASShareNFS(),
```
Comment (dariusbakunas) [inline truenas/provider.go:38]: could you run `go fmt ./...`, I was thinking to add git hook for that, but never got a chance

File: truenas/resource_truenas_share_smb_test.go [hunk 2]
```diff
@@ -0,0 +1,312 @@
...
+	"testing"
+)
+
+func TestAccResourceTruenasShareSMB_basic(t *testing.T) {
+	resourceName := "truenas_share_smb.smb"
+
+	var share api.ShareSMB
+
+	suffix := acctest.RandStringFromCharSet(5, acctest.CharSetAlphaNum)
+	dataset_name := fmt.Sprintf("%s-%s", testResourcePrefix, suffix)
```
Comment (dariusbakunas) [inline truenas/resource_truenas_share_smb_test.go:21]: totally minor, but I'd keep var naming consistent, ie `datasetName`

File: truenas/provider.go (hunk 1 above)
Comment (lkubb) [reply truenas/provider.go:38]: Whoops, forgot `./...`. Sorry I didn't catch that.

File: truenas/resource_truenas_share_smb_test.go (hunk 2 above)
Comment (lkubb) [reply truenas/resource_truenas_share_smb_test.go]: Good catch.

**Review by dariusbakunas — APPROVED** [review]

**Review by dariusbakunas — CHANGES_REQUESTED** [review]
not seeing go.mod changes, I think you may need to update that to pull in latest client, no? (nvm, looks like it already has latest)

**Review by dariusbakunas — APPROVED** [review]

**Pull request merged**